gtfs_funnel/schedule_stats_by_route_direction.py

```diff
@@ -40,28 +128,44 @@ def assemble_scheduled_trip_metrics(
     trip_cols = ["schedule_gtfs_dataset_key", "trip_instance_key"]
     
     grouped_df = df.groupby(trip_cols, observed=True, group_keys=False)
-
+    
+    trips_to_route_cols_subset = ["trip_instance_key", "route_id", "direction_id"]
+    
     # Get median / mean stop meters for the trip
     # Attach time-of-day and route_id and direction_id
-    df2 = pd.merge(
+    # Merge using a subset
+    median_stop_meters_df= pd.merge(
         grouped_df.agg({"stop_meters": "median"}).reset_index().rename(
             columns = {"stop_meters": "median_stop_meters"}),
         time_of_day,
         on = "trip_instance_key",
         how = "left"
     ).merge(
-        trips_to_route,
+        trips_to_route[trips_to_route_cols_subset],
         on = "trip_instance_key",
         how = "inner"
     )
     
-    return df2
+    median_stop_meters_df.direction_id = median_stop_meters_df.direction_id.fillna(0)
```

Comment:
What if there are nulls in `cardinal_direction_df`? would this NaN filled in prevent some merges?

Reply:
I filled out any nan values in `direction_id` as 0 in `cardinal_direction_df`, so I think it should be ok?